chainntnfs/height_hint_cache.go

```diff
@@ -35,6 +35,15 @@ var (
 	ErrConfirmHintNotFound = errors.New("confirm hint not found")
 )
 
+// Config contains the HeightHintCache configuration
+type Config struct {
```

Comment:
Nit: since the config only applies to the cache, we could rename this to `CacheConfig` and rename the bool below to just `QueryDisabled`.